app/Services/Model/Imp/SummitPromoCodeService.php

```diff
@@ -231,6 +231,23 @@ public function updatePromoCode(Summit $summit, $promo_code_id, array $data, Mem
 
             }
 
+            // tags
+            if (isset($data['tags']) && isset($data['track_id'])) {
+                $track = $summit->getPresentationCategory(intval($data['track_id']));
+                if (is_null($track))
+                    throw new EntityNotFoundException(sprintf("Track with id %s does not belongs to summit id %s.", $data['track_id'], $summit->getId()));
+
+                $promo_code->clearTags();
+
+                foreach ($data['tags'] as $tag_value) {
+                    $tag = $track->getAllowedTagByVal($tag_value);
```

Comment:
also if the tag does not exists on tag repository should be created

Reply:
done